src/libsyntax/ptr.rs

```diff
@@ -65,6 +65,10 @@ impl<T: 'static> P<T> {
     {
         f(*self.ptr)
     }
+    /// Equivalent to and_then(|x| x)
+    pub fn unwrap(self) -> T {
```

Comment:
Wow I'm years late to this but see https://github.com/rust-lang/rust/pull/54277#discussion_r257181754